### DISCUSSION TITLE
RateML 2023

### DISCUSSION THREAD

**Discussion by maedoc** [the-virtual-brain/tvb-root#638] [Ideas]
Thanks @DeLaVlag  for initiating the a discussion  about the future of RateML at the beginning of the year.

We could discuss some issues/wishes that have been mentioned before:

- move it closer to native LEMS
- Make it more dynamic using template per distinguishable model section
- Adding functionality
  - accommodation for larger models 
  - targeting other backends
- The dependency of TVB ontology. 

two points in the issue at https://req.thevirtualbrain.org/browse/TVB-3063
- fix state variable limits in generated TVB models?
- not committing generated code to Git repo

Feel free to add other points @i-Zaak @sdiazpier @aperezmartin

**Comment by maedoc** [comment]
There may be potential interaction with code generation for data fitting with PyMC3/PyTorch if @dionperd or @JohnGriffiths want to participate?

**Comment by DeLaVlag** [comment]
For the closer to native LEMS discussion: I have set up a repo using LEMS definitions (https://github.com/the-virtual-brain/tvb-root/tree/rateml_lems/tvb_library/tvb/rateML) with the minval, maxval field additions as discussed here (https://github.com/LEMS/jLEMS/issues/102), following the showcase (https://github.com/OpenSourceBrain/TheVirtualBrainShowcase/tree/master/TVB_LEMS)

And for the dynamic templating, is this in the direction of that: https://github.com/the-virtual-brain/tvb-root/tree/master/tvb_library/tvb/simulator/backend/templates ?
